cpp/Platform.Converters/CachingConverterDecorator.h

```diff
@@ -1,15 +1,28 @@
-﻿namespace Platform::Converters
+﻿#pragma once
+#include <functional>
+#include <memory>
+#include <unordered_map>
+
+namespace Platform::Converters
 {
     template <typename ...> class CachingConverterDecorator;
-    template <typename TSource, typename TTarget> class CachingConverterDecorator<TSource, TTarget> : public IConverter<TSource, TTarget>
+    template <typename TSource, typename TTarget> class CachingConverterDecorator<TSource, TTarget>
```

Comment:
We might have only single function type as a type parameter. TSource and TTarget can be inferred from the callable object type.

Reply:
Yes, but we not might dynamically call them

Reply:
@Konard, Please, show this magic. There is no way to do this in C++. 
In extreme case, use deduction guides.